core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -685,7 +685,7 @@ class TransactionsTest extends IntegrationTestHarness {
       // get here without having bumped the epoch. If bumping the epoch is possible, the producer will attempt to, so
       // check there that the epoch has actually increased
       producerStateEntry =
-        brokers(partitionLeader).logManager.getLog(new TopicPartition(testTopic, 0)).get.producerStateManager.activeProducers(producerId)
+        brokers(partitionLeader).logManager.getLog(new TopicPartition(testTopic, 0)).get.producerStateManager.activeProducers.get(producerId)
```

Comment:
One difference in behavior is that Java's `Map.get` returns `null` if no mapping is found while Scala's `Map.apply` throws an exception. There's another similar example in this file.

Reply:
Added a not null check for `producerStateEntry`.

Reply:
aah, forgot to update the comment. I added that check initially, but later when I checked the old code, it did not have this kind of check in the old Scala code. I plan to add that as a followup, limiting this change mostly related to the refactoring.

Reply:
It's fine for now since the test will still fail if no value is returned for the key, but it will result in a NPE instead of a NoSuchKeyException.